### ISSUE TITLE
Make repo public and push to npm

### ISSUE THREAD

**Issue by mikkoh** [Shopify/eslint-test-generator#1]
@minasmart and @yomexzo could you review this? This will be used for the `js-buy-sdk` to generate tests based on linting.

I think one thing that would be nice to do would be to make `eslint` be a peer dep. Because most likely you'll want use the exact same version of `eslint` as what your project is using. Thoughts? The annoying thing is that peer deps dont get auto installed now. So even if you use https://github.com/Shopify/babel-plugin-eslint-test-generator you'd have to explicitly install `eslint`

**Comment by minasmart** [comment]
Missing module declaration: https://github.com/Shopify/eslint-test-generator/blob/master/src/templates/qunit.js

**Comment by minasmart** [comment]
Shebang on [this file](https://github.com/Shopify/eslint-test-generator/blob/master/src/cli.js) doesn't make sense to me. That file is not executable by any version of node.

**Comment by minasmart** [comment]
https://github.com/Shopify/eslint-test-generator/blob/master/src/cli.js#L10-L16
`path` is only defined once. Use `const`. `template` is at one point a file name, and at one point a file body. This means that your state mutation is actually changing what _type_ of data the name refers to, which is a sort of hinky use of reassignment/naming to me.

**Comment by mikkoh** [comment]
Thanks for reviewing this @minasmart. So here's some comments to the above:

### Module Declaration

I realize it's missing. I'm sort of unsure whether it's needed since we'll be injecting these Unit tests at the end of the program. Typically I'd imagine that unit tests would live in one program so all globals and everything would all be there. When I wrote this module for Mocha I was initially requiring in assert etc.

### Shebang

Yep this code will never run in versions of Node. This file get's transpiled into the lib directory and the `bin` property in `package.json` points to `lib/cli.js`

https://github.com/Shopify/eslint-test-generator/blob/master/package.json#L42

### `template`

So ideally for this module (CLI) we'd like to be able to pass either a filename/path or a constant which defines which template we'll use. I think in order to allow for this and to not have reassignment you'd have two variables and always prefer the "handlebars template string" over the template type. I think this is sort of redundant and would cause complexity throughout the rest of the code for no reason.

**Comment by minasmart** [comment]
OK, so

> Module Declaration

Injecting these tests at the end, just means that they'll run as the last module. So if my last module is `someFunction behaves according to some spec`, then a bunch of eslint tests will run under that module declaration. That's probably bad.

> Sheband
> cool.
> 
> `template`
> I'd argue that it's harder to reason about the way it is, because I don't know if the template  is a compiled template, or a path, or what.

``` javascript
let templateString;

if (!SUITES[template]) {
  templateString = fs.readFileSync(template, 'utf8');
} else {
  templateString = ...
}

console.log(api({
  templateString,
  paths: paths.split(','),
}));
```

There's a clear boundary between what the argument represents, and what the var name represents at different points in time. I find the current approach obfuscates the identity of that name `template`.

**Comment by mikkoh** [comment]
@minasmart valid points.

If we `export` the tests as a module how would you require in? There's no file that gets generated and I wonder if having a global module name that you could import/require in would be wierd because there might be collisions with existing modules.

What would you do with the `template` option? I agree it is a bit confusing that when running via the `cli` its a path to a file, or a hardcoded test suite name. When using via API it can be a hardcoded test suite name or a `handlebars` template string. Would it make sense if we just made it so that it can always be all the the things? eg.

```
if(does the name exist in the list of test suites) {
... use test suite
} else if(check if it's a path if so read) {
... use read in template
} else {
... it wasn't a test suite name or a path to a file so we'll assume its a template string
}
```

**Comment by minasmart** [comment]
module as in the qunit `module('My Test Suite`)` module.

**Comment by minasmart** [comment]
Regarding the template string, I'd rather do some early introspection to figure out what it is. Hand it off to a normalization function, then assign the results of that normalization to a new name. So we know we have some weirdo thing that could be anything, but this new name represents a thing of a known type.

**Comment by mikkoh** [comment]
Ok @minasmart I've moved resolving `opts.template` to https://github.com/Shopify/eslint-test-generator/blob/master/src/resolve-template.js

This is kind of nice actually... Cause for instance if we wanted cache contents of file we could handle it here.

I ended up killing the CLI version of this generator. It was just pushing this module out of scope. (would have had to start parsing eslint options in command line and then passing to eslint)

I added a an import statement for qunit

https://github.com/Shopify/eslint-test-generator/blob/master/src/templates/qunit.js
